staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go

```diff
@@ -314,6 +314,23 @@ func (o *DrainCmdOptions) RunDrain() error {
 			}
 			break
 		}
+		if o.drainer.DryRun {
+			list, errs := o.drainer.GetPodsForDeletion(info.Name)
```

Comment:
I'm inclined to say we should move this logic into `o.deleteOrEvictPodsSimple(info)` which is invoked earlier just to not repeat the code below.

Reply:
i've moved the logic, ready for another look